tests/unit/layers/models/layersManager.spec.ts

```diff
@@ -0,0 +1,717 @@
+    });
+
+    it('pass if layer status is completed', async function () {
+      const taskParams = [
+        {
+          discreteId: 'testid1',
+          version: '1.0',
+          fileNames: ['file1.test1'],
+          originDirectory: 'test1-dir',
+          minZoom: 1,
+          maxZoom: 1,
+          layerRelativePath: layerRelativePath,
+          bbox: [0, 0, 90, 90],
+        },
+      ];
+
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1' });
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(undefined);
+      catalogExistsMock.mockResolvedValue(false);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([{ status: OperationStatus.COMPLETED }]);
+      generateTasksParametersMock.mockReturnValue(taskParams);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      // layersManager = new LayersManager(
+      //   logger,
+      //   zoomLevelCalculator,
+      //   jobManagerClientMock,
+      //   catalogClientMock,
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      const action = async () => {
+        await layersManager.createLayer(testData, managerCallbackUrl);
+      };
+      await expect(action()).resolves.not.toThrow();
+    });
+
+    it('pass if layer status is failed', async function () {
+      const taskParams = [
+        {
+          discreteId: 'testid1',
+          version: '1.0',
+          fileNames: ['file1.test1'],
+          originDirectory: 'test1-dir',
+          minZoom: 1,
+          maxZoom: 1,
+          layerRelativePath: layerRelativePath,
+          bbox: [0, 0, 90, 90],
+        },
+      ];
+
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1' });
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(undefined);
+      mapExistsMock.mockResolvedValue(false);
+      catalogExistsMock.mockResolvedValue(false);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([{ status: OperationStatus.FAILED }]);
+      generateTasksParametersMock.mockReturnValue(taskParams);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
```

Comment:
why the remarks?

Reply:
moved into before each test instad of create new instance every test manualy